gc/base/MemorySubSpace.cpp

```diff
@@ -919,17 +919,51 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 	}
 
 	if (_collector && _usesGlobalCollector) {
-		/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
-		_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		bool invokeGC = true;
+		if (J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode ) {
+			MM_MemorySpace* defaultMemorySpace = _extensions->heap->getDefaultMemorySpace();
+			uintptr_t freeMemorySize = defaultMemorySpace->getApproximateActiveFreeMemorySize(MEMORY_TYPE_OLD);
+			uintptr_t actvMemorySize = defaultMemorySpace->getActiveMemorySize(MEMORY_TYPE_OLD);
+			uintptr_t previousMemorySize;
```

Comment:
I consciously left it uninitialised as I previously received a comment related to initialisation.

dmitripivkine 8 days ago
again, no need to set variable to 0 if next line set it's real value. For the following scenario.

+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
 +uintptr_t
 +MM_MemoryPoolLargeObjects::releaseFreeMemoryPages(MM_EnvironmentBase* env)
 +{
 +	uintptr_t memReleased = 0;

did i understand it correctly? If not, apologise, please correct me.